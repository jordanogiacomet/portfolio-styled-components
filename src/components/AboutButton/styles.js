import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
