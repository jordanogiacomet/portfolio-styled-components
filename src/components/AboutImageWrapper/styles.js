import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
