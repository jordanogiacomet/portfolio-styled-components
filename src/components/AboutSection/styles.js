import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
