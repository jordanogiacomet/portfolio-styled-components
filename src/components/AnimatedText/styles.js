import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;