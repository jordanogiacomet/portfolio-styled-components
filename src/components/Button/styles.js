import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 