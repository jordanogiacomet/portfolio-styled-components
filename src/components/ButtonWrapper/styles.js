import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;