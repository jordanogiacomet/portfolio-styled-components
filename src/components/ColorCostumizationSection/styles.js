// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
