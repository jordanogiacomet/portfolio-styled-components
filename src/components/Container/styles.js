import styled, { css } from 'styled-components';

const containerStyles = {
  'card-container': css`
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    gap: 5rem;
  `,
  'home-container': css`
    width: 100%;
    margin-top: 10rem;
    margin-left: auto;
    margin-right: auto;
    padding: 1rem 3rem;
    z-index: 1;

    @media ${({ theme }) => theme.media.sm} { 
      max-width: ${({ theme }) => theme.container.maxWidth.sm};
    };

    @media ${({ theme }) => theme.media.md} { 
      max-width: ${({ theme }) => theme.container.maxWidth.md};
    };

    @media ${({ theme }) => theme.media.lg} { 
      max-width: ${({ theme }) => theme.container.maxWidth.lg};
    };

    @media ${({ theme }) => theme.media.xl} { 
      max-width: ${({ theme }) => theme.container.maxWidth.xl};
    };

    @media ${({ theme }) => theme.media.xxl} { 
      max-width: ${({ theme }) => theme.container.maxWidth.xxl};
    };
  `,
  'nav-container': css`
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
    padding: 1rem 2rem;
    margin: 1rem;
  `,
  'tab-container': css`
    display: flex;
    flex-direction: row;
    margin-top: 8px;
    margin-bottom: 25px;
`,
'radar-chart-container': css`
display: flex;
justify-content: center;
align-items: center;
margin-top: 30px;
@media ${({ theme }) => theme.media.md} {
  order: 1;
  margin-top: 0;
}
`,
  'tag-container': css`
    display: flex;
    justify-content: center;
    gap: 1rem;
    padding: 5rem;
  `,
  'text-container-about': css`
  margin-top: 30px;
  margin-bottom: 30px;
  @media ${({ theme }) => theme.media.md} {
    margin-top: 0;
    order: 2;
  }
  text-align: left;
  display: flex;
  flex-direction: column;
  height: auto;
`,
  'text-container-projects': css`
    margin-top: 30px;
    margin-bottom: 30px;
    @media ${({ theme }) => theme.media.md} {
      margin-top: 0;
    }
    text-align: left;
    display: flex;
    flex-direction: column;
    order: 1;
    height: auto;
  `,
  'form-container': css`
    margin-bottom: 2rem;
  `,
};

export const Container = styled.div`
  ${({ type }) => containerStyles[type] || ''};
`;