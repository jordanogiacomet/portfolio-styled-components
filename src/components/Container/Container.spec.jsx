import { screen } from '@testing-library/react';
import { Container } from '.';
import '@testing-library/jest-dom/extend-expect';
import { renderTheme } from '../../styles/render-theme';

describe('<Container />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<Container type="card-container">Children</Container>);
    expect(container).toMatchSnapshot();
  });

  it('should render Container with correct styles for card-container', () => {
    renderTheme(<Container type="card-container">Children</Container>);
    const container = screen.getByText('Children');
    expect(container).toHaveStyle(`
      display: flex;
      flex-wrap: wrap;
      justify-content: center;
      gap: 5rem;
    `);
  });

  it('should render Container with correct styles for home-container', () => {
    renderTheme(<Container type="home-container">Children</Container>);
    const container = screen.getByText('Children');
    expect(container).toHaveStyle(`
      width: 100%;
      margin-top: 8rem;
      margin-left: auto;
      margin-right: auto;
      padding: 1rem 3rem;
    `);
  });
});
