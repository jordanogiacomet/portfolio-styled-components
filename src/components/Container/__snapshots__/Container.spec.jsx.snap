// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match snapshot 1`] = `
.c0 {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 5rem;
}

<div>
  <div
    class="c0"
    type="card-container"
  >
    Children
  </div>
</div>
`;
